lib/apiml.js

```diff
@@ -202,6 +203,93 @@ ApimlConnector.prototype = {
     });
   },
   
+  getInstanceId() {
+    return this.zluxServerEurekaClient.config.instance.instanceId;
+  },
+  
+  getZluxInstances() {
+    const eurekaClient = this.zluxServerEurekaClient;
+    return new Promise((resolve, reject) => {
+      eurekaClient.once('registryUpdated', () => {
+        const zluxInstances = eurekaClient.getInstancesByVipAddress('ZLUX');
+        log.debug(`registry updated, zluxInstances ${JSON.stringify(zluxInstances, null, 2)}`);
+        zluxInstances.sort((a, b) => a.instanceId.localeCompare(b.instanceId));
+        resolve(zluxInstances);
+      });
+    });
+  },
+  
+  waitUntilZluxClusterIsReady(clusterSize) {
+    return this.getZluxInstances().then(
+      zluxInstances => {
+        if (zluxInstances.length < clusterSize) {
+          return this.waitUntilZluxClusterIsReady(clusterSize);
```

Comment:
what does the word cluster mean here?
does it mean the app-server that is running in cluster mode and has a number of workers?
or does it mean a group of multiple app servers?

Reply:
if this is about the number of workers, i think the code must be changed to only need 1 eureka client per app server, by using the cluster master to handle it.

Reply:
It is about a number of zlux-app-server instances. Yes, I should give it more apt name.